src/java/us/kbase/templates/module_python_impl.vm.properties

```diff
@@ -79,109 +63,150 @@ class ${module_name}:
+        report += 'Number of initial contigs:      '+ str(n_total) + '\n'
+        report += 'Number of contigs removed:      '+ str(n_total - n_remaining) + '\n'
+        report += 'Number of contigs in final set: '+ str(n_remaining) + '\n'
 
-        # Step 5 - Build a Report and return
         reportObj = {
-            'objects_created': [{'ref': new_assembly, 'description': 'Filtered contigs'}],
-            'text_message': 'Filtered Assembly to ' + str(n_remaining) + ' contigs out of ' + str(n_total)
+            'objects_created':[{
+                    'ref':str(info[6]) + '/'+str(info[0])+'/'+str(info[4]), 
+                    'description':'Filtered Contigs'
+                }],
+            'text_message':report
         }
-        report = KBaseReport(self.callback_url)
-        report_info = report.create({'report': reportObj, 'workspace_name': params['workspace_name']})
-
-
-        # STEP 6: contruct the output to send back
-        output = {'report_name': report_info['name'],
-                  'report_ref': report_info['ref'],
-                  'assembly_output': new_assembly,
-                  'n_initial_contigs': n_total,
-                  'n_contigs_removed': n_total - n_remaining,
-                  'n_contigs_remaining': n_remaining
-                  }
-        logging.info('returning:' + pformat(output))
+
+        # generate a unique name for the Method report
+        reportName = 'filter_contigs_report_'+str(hex(uuid.getnode()))
+        report_info_list = None
+        try:
+            report_info_list = wsClient.save_objects({
+                    'id':info[6],
+                    'objects':[
+                        {
+                            'type':'KBaseReport.Report',
+                            'data':reportObj,
+                            'name':reportName,
+                            'meta':{},
+                            'hidden':1, # important!  make sure the report is hidden
+                            'provenance':provenance
+                        }
+                    ]
+                })
+        except:
+            exc_type, exc_value, exc_traceback = sys.exc_info()
+            lines = traceback.format_exception(exc_type, exc_value, exc_traceback)
+            orig_error = ''.join('    ' + line for line in lines)
+            raise ValueError('Error saving filtered ContigSet object to workspace:\n' + orig_error)
+        
+        report_info = report_info_list[0]
+
+        print('saved Report: '+pformat(report_info))
+
+        returnVal = {
+                'report_name': reportName,
+                'report_ref': str(report_info[6]) + '/' + str(report_info[0]) + '/' + str(report_info[4]),
+                'new_contigset_ref': str(info[6]) + '/'+str(info[0])+'/'+str(info[4]),
+                'n_initial_contigs':n_total,
+                'n_contigs_removed':n_total-n_remaining,
+                'n_contigs_remaining':n_remaining
+            }
```

Comment:
weird indentation?

Can't you just run ruff format over the file and then ensure that the `#BEGIN` and `#END` commands don't have a space after the `#`?